src/routes/messages/messages.service.ts

```diff
@@ -45,16 +45,18 @@ export class MessagesService {
     paginationData: PaginationData;
     routeUrl: Readonly<URL>;
   }): Promise<Page<DateLabel | MessageItem>> {
-    const safe = await this.safeRepository.getSafe({
-      chainId: args.chainId,
-      address: args.safeAddress,
-    });
-    const page = await this.messagesRepository.getMessagesBySafe({
-      chainId: args.chainId,
-      safeAddress: args.safeAddress,
-      limit: args.paginationData.limit,
-      offset: args.paginationData.offset,
-    });
+    const [safe, page] = await Promise.all([
```

Comment:
Not strictly related, but I saw this performance increase opportunity while reviewing the messages retrieval flow. We can do it separately if you prefer.

Reply:
In general I prefer to leave the changes within the scope of the PR. You can add this change to the commit message/PR description 👍

Reply:
Done, added to the PR description, thanks!